Content.Shared/Research/Systems/BlueprintSystem.cs

```diff
@@ -66,6 +68,9 @@ public bool TryInsertBlueprint(Entity<BlueprintReceiverComponent> ent, Entity<Bl
 
         var ev = new TechnologyDatabaseModifiedEvent();
         RaiseLocalEvent(ent, ref ev);
+
+        _materialStorage.UpdateMaterialWhitelist(ent);
+
```

Comment:
i think the best way to do this is just to subscribe to technologyDatabaseModifiedEvent with MaterialStorageComponent and call this function in there. That way, in the future, it works with any kind of recipe additions instead of just blueprints.

Reply:
Alright made that change. Was confused for a while when I was trying to add it into the file MaterialStorageComponent, instead of MaterialStorageSystem.

Reply:
> i think the best way to do this is just to subscribe to technologyDatabaseModifiedEvent with MaterialStorageComponent and call this function in there. That way, in the future, it works with any kind of recipe additions instead of just blueprints.

So would this mean that if we created a graph for a machine and the graph includes something the machine doesn't store now, it automatically whitelists the required materials for that?